### PR TITLE
CDPT-599 fix blank screen with date range search

### PR DESCRIPTION
**Pull request by njpreddy**
**Issue:**
Searching the database by date range resulted in a http 500 error page being displayed.

fum-date-range-search

fum-date-range-search-failure

This PR fixes this and returns expected search results

fum-date-range-search-sucess

**Jira ticket:**
https://dsdmoj.atlassian.net/jira/software/c/projects/CDPT/boards/864?modal=detail&selectedIssue=CDPT-599

### CODE REVIEW

File: app/models/search.rb [hunk 1]
```diff
@@ -12,27 +12,13 @@ class Search
...
-    day = value[3]
-    @date_to = begin
-      Date.new(year, month, day)
-    rescue TypeError, Date::Error
-      value
-    end
+    build_date(value, "to")
   end
 
   def results
```
Comment (vertism) [inline app/models/search.rb:24]: Maybe worth adding a test for what was failing to protect against a future regression.

File: app/models/search.rb [hunk 2]
```diff
@@ -73,4 +54,24 @@ def validate_dates
       errors.delete(:keywords)
     end
   end
+
+  def build_date(value, field)
+    year = value[1]
+    month = value[2]
+    day = value[3]
+
+    if field == "from"
```
Comment (vertism) [inline app/models/search.rb:63]: Nice idea to make this into a method. 
You could avoid the if condition by returning the output and assigning when you call the method. 
e.g.
```
@date_from = build_date(value)
```

File: config/locales/en.yml [hunk 3]
```diff
@@ -30,7 +30,7 @@ en:
         search:
           attributes:
             keywords:
-              blank: Keywords can't be blank
+              blank: Please enter Keywords and/or a valid date range
```
Comment (vertism) [inline config/locales/en.yml:33]: Given what Ian was saying about not making the service too easy to use, is there an argument that we shouldn't be highlighting that you can search with just a date range?

File: app/models/search.rb (hunk 1 above)
Comment (njpreddy) [reply app/models/search.rb:24]: I've added tests for date range searches which do and do not return results.

File: app/models/search.rb (hunk 2 above)
Comment (njpreddy) [reply app/models/search.rb]: Great! thanks for that.  Implemented as per your suggestion.

File: config/locales/en.yml (hunk 3 above)
Comment (njpreddy) [reply config/locales/en.yml]: Agreed, I've reverted this text change.

File: spec/system/search_spec.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 RSpec.describe "Search" do
   before do
-    create(:case, case_name: "Test Case")
+    # create(:case, case_name: "Test Case")
```
Comment (vertism) [inline spec/system/search_spec.rb:3]: Presumably this can be removed
```suggestion
```

**Review by vertism — APPROVED** [review]
Looks good 👍

File: spec/system/search_spec.rb (hunk 4 above)
Comment (njpreddy) [reply spec/system/search_spec.rb]: Done!

**Pull request merged**